pkg/transform/transform.go

```diff
@@ -0,0 +1,377 @@
+// Copyright (c) Bartłomiej Płotka @bwplotka
+// Licensed under the Apache License 2.0.
+
+package transform
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/bwplotka/mdox/pkg/mdformatter"
+	"github.com/efficientgo/tools/pkg/errcapture"
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v3"
+)
+
+// Dir transforms directory using given configuration file.
+func Dir(ctx context.Context, logger log.Logger, configFile string) error {
+	c, err := parseConfigFile(configFile)
+	if err != nil {
+		return err
+	}
+
+	for _, d := range []string{c.OutputDir, c.OutputStaticDir} {
+		if d == "" {
+			continue
+		}
+
+		_, err = os.Stat(d)
+		if err != nil && !os.IsNotExist(err) {
+			return err
+		}
+
+		if !os.IsNotExist(err) {
+			if err := os.RemoveAll(d); err != nil {
+				return err
+			}
+		}
+		if err := os.MkdirAll(d, os.ModePerm); err != nil {
+			return err
+		}
+
+		if c.GitIgnored {
+			if err = ioutil.WriteFile(filepath.Join(d, ".gitignore"), []byte("*.*"), os.ModePerm); err != nil {
```

Comment:
I think gitignore can just be `*` instead of `*.*`. This would work better if there is non-md file without extension, for example `Procfile`. 🙂

Reply:
👍🏽